### DISCUSSION TITLE
Proposal: "yield all elements" statement

### DISCUSSION THREAD

**Discussion by masonwheeler** [dotnet/csharplang#8351] [Language Ideas]
This isn't *incredibly* common, but it pops up often enough when working with enumerables, that it would be nice to have a simpler way to write it:
```
foreach (var foo in bar)
{
    yield return foo;
}
```

Would it be possible to get a way to represent this pattern as a single statement that would lower to the loop above?  Something like `yield in bar;` or `yield from bar;`, for example.

**Comment by 333fred** [comment]
Dupe of https://github.com/dotnet/csharplang/discussions/378.
